other/js/websockify.js

```diff
@@ -161,7 +162,23 @@ if (argv.cert) {
     console.log("    - Running in unencrypted HTTP (ws://) mode");
     webServer = http.createServer(http_request);
 }
+
+if (argv["auth-plugin"]) {
+    let auth_plugin_arg = argv["auth-plugin"].split(".")
+    let plugin_name = auth_plugin_arg.pop();
+    let module_path = auth_plugin_arg.join(".");
+
+    let auth_plugin = require(module_path)[plugin_name];
+    let auth_source = argv["auth-source"] || undefined;
+    websocket_server_opts = {
+        server: webServer,
+        verifyClient: auth_plugin(auth_source)
```

Comment:
What about the case (that is symmetrical to the python version) where auth_plugin is a class function and not just a plain function returning the auth function?

I think it would actually be more idiomatic JavaScript to have the auth_source bound as the first argument of an authorizer function rather than generating a new function in the auth_plugin itself. For example:

```
verifyClient: auth_plugin.bind(auth_plugin, auth_source)
```

Obviously, the structure of your example authentication function would change (one less layer of abstraction).

Reply:
Addressed in 5c3ac52. The auth plugin can now either be a class with a `authenticate` method (similar to the Python version), or a factory function which returns an object with an `authenticate` method.